hera_cal/smooth_cal.py

```diff
@@ -206,30 +206,52 @@ def check_consistency(self):
+                #test time contiguity
+                assert(np.abs(np.median(np.diff(self.times)) - self.times[0] + self.prev_times[-1]) < 1e-6)
+                assert(np.abs(np.median(np.diff(self.times)) - np.median(np.diff(self.prev_times))) < 1e-6)
+                #assert(np.all(self.prev_times == self.prev_data_times)) #TODO: revisit this
+            except AssertionError:
+                warnings.warn('prev_cal and/or prev_data is inconsistent with central cal/data and will not be used for time smoothing.')
+                self.has_prev_cal, self.has_prev_data = False, False
+
+            # if prev_gain or prev_data is missing an antenna (e.g. because it was in xants and omnical dropped it), treat it as flagged 
             for (ant,pol) in self.gains.keys():
-                assert((ant,pol) in self.prev_gains.keys()) #assert prev_gains has all the same keys
-                assert((ant, ant, pol+pol) in self.prev_data) #assert data has autocorrelations
+                if (ant,pol) not in self.prev_gains.keys() or (ant, ant, pol+pol) not in self.prev_data.keys():
+                    self.prev_gains[(ant,pol)] = np.ones_like(self.prev_gains.values()[0]) # set gains to 1
+                    self.prev_flags[(ant,pol)] = np.ones_like(self.prev_flags.values()[0]) # all flagged
+                    self.prev_data[(ant, ant, pol+pol)]  = np.ones_like(self.prev_data.values()[0]) # set autos to 1
+                    self.prev_data_ant_flags[(ant,pol)]  = np.ones_like(self.prev_data_ant_flags.values()[0]) # all flagged
         
         if self.has_next_cal or self.has_next_data:
-            assert(self.has_next_cal and self.has_next_data) #should have both or it doesn't make sense
-            assert(np.all(self.next_freqs == self.freqs))
-            assert(np.all(self.next_data_freqs == self.freqs))
-            assert(len(self.next_times) == len(self.next_data_times))
-            #test time contiguity
-            assert(np.abs(np.median(np.diff(self.times)) + self.times[-1] - self.next_times[0]) < 1e-6)
-            assert(np.abs(np.median(np.diff(self.times)) - np.median(np.diff(self.next_times))) < 1e-6)
-            #assert(np.all(self.next_times == self.next_data_times)) #TODO: revisit this
+            try:
+                assert(self.has_next_cal and self.has_next_data) #should have both or it doesn't make sense
+                assert(np.all(self.next_freqs == self.freqs))
+                assert(np.all(self.next_data_freqs == self.freqs))
+                assert(len(self.next_times) == len(self.next_data_times))
+                #test time contiguity
+                assert(np.abs(np.median(np.diff(self.times)) + self.times[-1] - self.next_times[0]) < 1e-6)
+                assert(np.abs(np.median(np.diff(self.times)) - np.median(np.diff(self.next_times))) < 1e-6)
+                #assert(np.all(self.next_times == self.next_data_times)) #TODO: revisit this
+                for (ant,pol) in self.gains.keys():
```

Comment:
It seems like the corresponding for loop above was changed to be more flexible in terms of not requiring all antennas are consistent (i.e., the `assert` statements were replaced with an `if` statement). Should we also be less strict down here?

Reply:
Yes. Good call. Meant to delete this loop.